TWLight/emails/tasks.py

```diff
@@ -322,6 +324,33 @@ def update_app_status_on_save(sender, instance, **kwargs):
                     'but no such handler exists'.format(handler_key=handler_key))
             pass
 
+@receiver(pre_save, sender=AccessCode)
+def send_authorization_emails(sender, instance, **kwargs):
+    """
+    When access code objects are updated, check if they should trigger
+    an email to a user.
+    """
+
+    # The AccessCode should already exist, don't trigger on code creation.
+    if instance.id:
+        orig_code = AccessCode.objects.get(pk=instance.id)
+        # If this code is having an authorization added where it didn't
+        # have one before, we've probably just finalised an application
+        # and therefore want to send an email.
+        if not orig_code.authorization and instance.authorization:
+
+            base_url = get_current_site(None).domain
+
+            access_code_instructions = instance.partner.access_code_instructions
+            mail_instance = MagicMailBuilder()
+
+            email = mail_instance.access_code_email(instance.authorization.authorized_user.email,
+                {'editor_wp_username': instance.authorization.authorized_user.editor.wp_username,
+                 'partner': instance.partner,
+                 'access_code': instance.code,
+                 'body': access_code_instructions
```

Comment:
I feel like this should probably be `access_code_instructions` instead of `body` for legibility.
```suggestion
                 'access_code_instructions': access_code_instructions
```